src/vm.c

```diff
@@ -467,22 +483,40 @@ V7_PRIVATE val_t v7_property_value(struct v7_property *p) {
   return p->value;
 }
 
-V7_PRIVATE val_t v7_array_length(struct v7 *v7, val_t v) {
-  struct v7_property *prop;
+V7_PRIVATE long v7_array_length(struct v7 *v7, val_t v) {
```

Comment:
Please also fix unit tests that check for array length treating it as `val_t`, e.g.:
``` ASSERT(check_value(v7, v7_array_length(v7, v), "3"));```